12-proof-system-p.mm1

```diff
@@ -861,14 +868,14 @@ do {
     [$or ,phi1 ,phi2$      '(bitr membership_or_bi    @ cong_of_equiv_or    ,(propag_mem x phi1) ,(propag_mem x phi2))]
     [$and ,phi1 ,phi2$     '(bitr membership_and_bi   @ cong_of_equiv_and   ,(propag_mem x phi1) ,(propag_mem x phi2))]
     [$equiv ,phi1 ,phi2$   '(bitr membership_equiv_bi @ cong_of_equiv_equiv ,(propag_mem x phi1) ,(propag_mem x phi2))]
-    -- [$app ,phi1 ,phi2$     '(membership_app    ,(propag_mem x phi1) ,(propag_mem x phi2))]
+    [$app ,phi1 ,phi2$     '(bitr membership_app      @ cong_of_equiv_exists @ cong_of_equiv_and_l ,(propag_mem #undef phi2))]
```

Comment:
What does the `#undef` mean here?

Reply:
It's a marker that stands for undefined. It exists there because we're introducing a fresh variable "y" and propagating the membership with this new variable instead of the original "x" but we don't know the actual name of the variable that the user will want so we cannot use "y". Instead it's equally appropriate to use #undef so that all == comparisons down the line will fail, which is okay because this new "y" is disjoint from "phi".

Reply:
Thanks